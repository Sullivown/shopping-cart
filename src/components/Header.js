import React from 'react';
import { Link } from 'react-router-dom';
import logo from '../assets/images/logo.png';
import cartImg from '../assets/images/cart.png';

function Header(props) {
	return (
		<header>
			<div className='site-title-div'>
				<Link to='/'>
					<img src={logo} alt='Logo' />
				</Link>
				<Link to='/'>
					<h1>AdventureMart</h1>
				</Link>
			</div>
			<nav>
				<ul>
					<li>
						<Link to='about'>About</Link>
					</li>
					<li>
						<Link to='shop'>Shop</Link>
					</li>

					<li className='cart-container'>
						<Link to='shop/cart'>
							<img
								className='cart-img'
								src={cartImg}
								alt='Shopping cart'
							/>
							<span data-testid='total-basket-items'>
								{props.totalItems && props.totalItems}
							</span>
						</Link>
					</li>
				</ul>
			</nav>
		</header>
	);
}

export default Header;
