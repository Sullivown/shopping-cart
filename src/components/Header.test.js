import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { MemoryRouter } from 'react-router-dom';

import Header from './Header';

describe('Header component', () => {
	it('renders header', () => {
		render(<Header />, { wrapper: MemoryRouter });

		expect(screen.getByRole('banner')).toBeInTheDocument();
	});

	it('renders 5 nav links', () => {
		render(<Header />, { wrapper: MemoryRouter });

		expect(screen.getAllByRole('link').length).toBe(5);
	});

	it('renders zero basket items', () => {
		render(<Header totalItems={0} />, { wrapper: MemoryRouter });

		expect(screen.getByTestId('total-basket-items').textContent).toBe('0');
	});

	it('renders one basket item', () => {
		render(<Header totalItems={1} />, { wrapper: MemoryRouter });

		expect(screen.getByTestId('total-basket-items').textContent).toBe('1');
	});
});
