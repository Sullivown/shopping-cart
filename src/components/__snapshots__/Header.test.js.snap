// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders header 1`] = `
<div>
  <header>
    <div
      class="site-title-div"
    >
      <a
        href="/"
      >
        <img
          alt="Logo"
          src="logo.png"
        />
      </a>
      <a
        href="/"
      >
        <h1>
          AdventureMart
        </h1>
      </a>
    </div>
    <nav>
      <ul>
        <li>
          <a
            href="/about"
          >
            About
          </a>
        </li>
        <li>
          <a
            href="/shop"
          >
            Shop
          </a>
        </li>
        <li
          class="cart-container"
        >
          <a
            href="/shop/cart"
          >
            <img
              alt="Shopping cart"
              class="cart-img"
              src="cart.png"
            />
            <span
              data-testid="total-basket-items"
            />
          </a>
        </li>
      </ul>
    </nav>
  </header>
</div>
`;
